src/components/blog/table-of-contents.tsx

```
@@ -0,0 +1,122 @@
+export function TableOfContents() {
+  const [headings, setHeadings] = useState<Heading[]>([]);
+  const [activeId, setActiveId] = useState<string>("");
+
+  useEffect(() => {
+    // Extract headings from the page
+    const elements = document.querySelectorAll("h2, h3");
+    const headingData: Heading[] = [];
+    const seenIds = new Set<string>();
+
+    elements.forEach((element) => {
+      let baseId =
```

ESLint

[error] 'baseId' is never reassigned. Use 'const' instead.

(prefer-const)